hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/AssociatePrecompileSuite.java

```diff
@@ -316,7 +316,7 @@ private HapiSpec invalidSingleAbiCallConsumesAllProvidedGas() {
                             .getTransactionRecord(INVALID_SINGLE_ABI_CALL_TXN)
                             .getContractCallResult()
                             .getGasUsed();
-                    assertEquals(99011, gasUsed);
+                    assertEquals(99014, gasUsed);
```

Comment:
This number was found empirically, I assume?

Reply:
yes, the formula for calculating the gas cost is:
start with the base cost of 21k gas
add 4 gas for each zero in the input bytes
add 16 gas for each non-zero in the input bytes
add 32k gas if this is a creation
It’s all calculated in the [CustomGasCalculator](https://github.com/hashgraph/hedera-services/blob/d0e5885c155e92e69d3e49fb9ef4fe4fe76baa50/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/CustomGasCalculator.java#L50).
I think it's impractical to calculate it per test, but it's useful to have these assertions and checks to help us validate the values only change when it makes sense.